packages/amplify-graphql-schema-generator/src/__tests__/generate-ts-data-schema.test.ts

```diff
@@ -0,0 +1,61 @@
+import { Engine, Field, Model, Schema } from '../schema-representation';
```

Comment:
If I read this correctly, the unit & e2e tests are almost identical, except that the actual assertions differ: unit does a snapshot comparison, and e2e does a tsc, is that right? Any chance of refactoring the common elements to reduce duplication?

Reply:
Yes, that's right. Ideally I would like to extend this test to pull the SQL IR from a real database and generate the typebeast schema. I would like to take care of it in a follow up PR.